dockstore-client/src/main/java/io/dockstore/client/cli/nested/NotificationsClients/NotificationsClient.java

```diff
@@ -0,0 +1,117 @@
+package io.dockstore.client.cli.nested.NotificationsClients;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+import com.google.gson.Gson;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author gluu
+ * @since 12/01/18
+ */
+public class NotificationsClient {
+    public static final String PROVISION_INPUT = "provision-in";
+    public static final String RUN = "workflow-start";
+    public static final String PROVISION_OUTPUT = "provision-out";
+    public static final String COMPLETED = "workflow-complete";
+    protected static final String USERNAME = "Dockstore CLI";
+    private static final Logger LOG = LoggerFactory.getLogger(NotificationsClient.class);
+    protected String hookURL;
+    protected String uuid;
+
+
+    public NotificationsClient(String hookURL, String uuid) {
+        this.hookURL = hookURL;
+        this.uuid = uuid;
+    }
+
+    /**
+     * Send message using the webhook URL originally provided
+     *
+     * @param message Base message indicating the step of the tool/workflow
+     * @param success The status of the step
+     */
+    public void sendMessage(String message, boolean success) {
+        if (uuid == null || uuid.isEmpty()) {
+            return;
+        } else if (hookURL == null || hookURL.isEmpty()) {
+            System.err.println("Notifications UUID is specified but no notifications webhook URL found in config file");
```

Comment:
Good idea, but let's take it further and check for this condition on start-up and exit to make it more obvious

Reply:
Previously it checks every time a message to about to be sent.  Now it just checks once upon start-up (launch command).